main.py

```diff
@@ -54,14 +55,20 @@ def setup(self):
         self._lock = Lock()
 
         self.execute_as_loop(settings.DEPLOY_EVCS_INTERVAL)
+        self.load_time = time.time()
         self.load_all_evcs()
 
     def execute(self):
         """Execute once when the napp is running."""
         for circuit in tuple(self.circuits.values()):
             if circuit.is_enabled() and not circuit.is_active():
```

Comment:
```suggestion
            if circuit.is_enabled() and not circuit.is_active() and \
               not circuit.lock.locked():
```

This will avoid a circuit just created (under deployment) to request the lock again.